sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_custom_form_client.py

```diff
@@ -0,0 +1,374 @@
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff'. Other type of content is ignored.
+
+        :param str source: An Azure Storage blob container URI.
+        :param str source_prefix_filter: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob Uri, use the prefix to restrict sub
+            folders for training.
+        :param bool include_sub_folders: A flag to indicate if sub folders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed.
+        :return: LROPoller
+        :rtype: ~azure.core.polling.LROPoller
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        cls = kwargs.pop("cls", None)
+        response = self._client.train_custom_model_async(
+            train_request=TrainRequest(
+                source=source,
+                source_filter=TrainSourceFilter(
+                    prefix=source_prefix_filter,
+                    include_sub_folders=include_sub_folders
+                )
+            ),
+            cls=lambda pipeline_response, _, response_headers: pipeline_response,
+            **kwargs
+        )
+
+        def callback(raw_response):
+            model = self._client._deserialize(Model, raw_response)
+            return CustomModel._from_generated(model)
+
+        deserialization_callback = cls if cls else callback
+        return LROPoller(self._client._client, response, deserialization_callback, LROBasePolling(**kwargs))
+
+    @distributed_trace
+    def begin_labeled_training(self, source, source_prefix_filter="", include_sub_folders=False, **kwargs):
+        # type: (str, Optional[str], Optional[bool], Any) -> LROPoller
+        """Create and train a custom model with labels. The request must include a source parameter that is an
+        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff'. Other type of content is ignored.
+
+        :param str source: An Azure Storage blob container URI.
+        :param str source_prefix_filter: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob Uri, use the prefix to restrict sub
+            folders for training.
+        :param bool include_sub_folders: A flag to indicate if sub folders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed.
+        :return: LROPoller
+        :rtype: ~azure.core.polling.LROPoller
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        cls = kwargs.pop("cls", None)
```

Comment:
Just a comment, not sure on my end either, but what if the user specifically doesn't want a callback. If they pass None, we use the default cls. Not even sure this is valid behavior but I also have this question for the ta client so posting here

Reply:
discussed offline